c10/cuda/CUDACachingAllocator.h

```diff
@@ -208,7 +209,15 @@ class CUDAAllocator : public DeviceAllocator {
   virtual void* raw_alloc(size_t nbytes) = 0;
   virtual void* raw_alloc_with_stream(size_t nbytes, cudaStream_t stream) = 0;
   virtual void raw_delete(void* ptr) = 0;
-  virtual void init(int device_count) = 0;
+
+  // Any classes overriding init should call CUDAAllocator::init
+  virtual void init(int device_count) {
```

Comment:
Will this always run even if it is a no-op? e.g., setting memory fraction to 1.0?

Also curious if somehow it is changing the current CUDA device to the last one